src/V3Mutex.h

```diff
@@ -0,0 +1,152 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Thread pool for Verilator itself
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2005-2023 by Wilson Snyder.  This program is free software; you can
+// redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+// SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
+//
+//*************************************************************************
+
+#ifndef VERILATOR_V3MUTEX_H_
+#define VERILATOR_V3MUTEX_H_ 1
+
+#include "verilatedos.h"
+
+#include <cstdlib>
+#include <functional>
+#include <mutex>
+
+#define VL_LOCK_SPINS 50000  /// Number of times to spin for a mutex before yielding
+
+// MutexConfig class that allows to configure how mutex and lockgurads behave
+// once configured and locked, it cannot be changed. Configuration and lock needs to be
+// done before starting any additional threads.
+class V3MutexConfig final {
+private:
+    // Allows to disable mutexes and lockguards.
+    // Use carefully as it can cause undefined behavior when used inappropriately.
+    // All mutexes needs to be unlocked.
+    bool m_enable = false;  // Allows locking on mutexes, default don't lock mutexes
+    bool m_lockConfig = false;  // After set, configuration cannot be changed
+
+    V3MutexConfig() = default;
+    ~V3MutexConfig() = default;
+
+public:
+    static V3MutexConfig& s() VL_MT_UNSAFE {
+        static V3MutexConfig s;
+        return s;
+    }
+
+    // configures class
+    void configure(bool enable) VL_MT_UNSAFE {
+        if (!s().m_lockConfig) {
+            s().m_enable = enable;
+            s().m_lockConfig = true;
+        } else {
+            // requires <iostream>
+            // avoided to reduce compile time
+            // std::cerr << "%Error: V3Mutex already configured." << std::endl;
+            std::abort();
+        }
+    }
+    // get lockConfig value
+    bool lockConfig() VL_MT_SAFE { return m_lockConfig; }
+    // get enable value
+    bool enable() VL_MT_SAFE { return m_enable; }
```

Comment:
```suggestion
    bool lockConfig() VL_MT_SAFE { return m_lockConfig; }
    bool enable() VL_MT_SAFE { return m_enable; }
```

Or capitalize "get".  But seems obvious what they do